examples/template_engine.du

```diff
@@ -0,0 +1,58 @@
+import System;
+
+const LEFT_BRACE = "{";
+const RIGHT_BRACE = "}";
+
+// Template is a simple template rendering engine.
+class Template {
+    private tmpl;
+    private klass;
+    private rendered;
+
+    init(tmpl, klass) {
+        this.tmpl = tmpl;
+        this.klass = klass;
+    }
+
+    // render parses the given template and class and
+    // matches the fields in the class to the template
+    // fields and replaces them with the class values.
+    render() {
+        const classAttrs = this.klass.getAttributes()["properties"];
+        this.rendered = this.tmpl;
+
+        classAttrs.forEach(def(attr) => {
+            const attrVal = this.klass.getAttribute(attr);
+            const tmplField = LEFT_BRACE+attr+RIGHT_BRACE;
+
+            if (this.rendered.contains(tmplField)) {
+                if (type(attrVal) != "string") {
+                    this.rendered = this.rendered.replace(tmplField, attrVal.toString());
+                } else {
+                    this.rendered = this.rendered.replace(tmplField, attrVal);
+                }
+            }
+        });
+
+        return this.rendered;
+    }
+}
```

Comment:
Here I've done:
- Changed `this.rendered` to be variable `rendered` local to `render()`
- Changed the formatting for `tmplField` to use `.format` with the new class variables

```suggestion
    render() {
        const classAttrs = this.klass.getAttributes()["properties"];
        var rendered = this.tmpl;

        classAttrs.forEach(def(attr) => {
            const attrVal = this.klass.getAttribute(attr);
            const tmplField = "{}{}{}".format(
                Template.LEFT_BRACE,
                attr,
                Template.RIGHT_BRACE
            );

            if (rendered.contains(tmplField)) {
                if (type(attrVal) != "string") {
                    rendered = rendered.replace(tmplField, attrVal.toString());
                } else {
                    rendered = rendered.replace(tmplField, attrVal);
                }
            }
        });

        return rendered;
    }
```